FT2023-61-mvc

Rajdip-innoraft755 opened the pull request.

Rajdip-innoraft755 committed:
FT2023-61-mvc:create the basic mvc structure.

Rajdip-innoraft755 committed:
testing

Rajdip-innoraft755 committed:
testing.

Rajdip-innoraft755 committed:
add readme file in main.

Rajdip-innoraft755 committed:
FT2023-61-mvc:add models folder in application.

Rajdip-innoraft755 committed:
FT2023-61:create a login,signup page and store user data in DB , make validation of those pages, initialize the composer in the project folder.

Eshakundu01 commented on mvc/system/classes/frameWork.php:
```diff
@@ -1,20 +1,26 @@
 class frameWork
 {
-
 	public function view($viewName)
```
Preferable to use two spaces for indentation

Rajdip-innoraft755 replied:
changes done .

Eshakundu01 commented on mvc/system/classes/rout.php:
```diff
@@ -1,9 +1,9 @@
 class rout
 {
-  public $controller = "welcome";
+  public $controller = "home";
```
We can use protected as well since we don't need to access these variables outside the class.

Eshakundu01 commented on mvc/system/classes/rout.php:
```diff
@@ -15,18 +15,17 @@ public function __construct()
+  
   public function url()
   {
     if (isset($_SERVER["REQUEST_URI"])) {
```
In line number 33 we can use `trim` function instead of ` rtrim ` function and include the "/" as the removable character. Cause in line 35 when using substring if an extra "/" is provided it will be considered thus resulting in an error, so we can just explode the URL rather than substring.

Eshakundu01 commented on mvc/system/init.php:
```diff
@@ -1,8 +1,7 @@
 <?php
-	spl_autoload_register(function($className){
```
Please rectify the spacing in the code

Eshakundu01 commented on mvc/public/contact.php:
The deleted files should be removed from the repository also.

Rajdip-innoraft755 replied:
changes done.

Eshakundu01 commented on mvc/public/index.php:
```diff
@@ -2,6 +2,7 @@
 	ini_set('display_startup_errors', 1);
 	error_reporting(E_ALL);
+	session_start();
+	include "../config/config.php";
```
It is preferable to use `require_once` or `require` instead of include.

Eshakundu01 commented on mvc/system/init.php:
```diff
@@ -1,8 +1,7 @@
+spl_autoload_register(function ($className) {
+	include "classes/$className.php";
+});
+$rout = new rout();
```
Braces are not necessary to use here while initializing the object.

Eshakundu01 commented on mvc/system/classes/connectDB.php:
```diff
@@ -0,0 +1,22 @@
+		}
+	}
+
+	public function query($sql){
```
Use a similar bracket format throughout the code.

Eshakundu01 commented on mvc/config/config.php:
```diff
@@ -0,0 +1,14 @@
+	define("USERNAME","root");
+	define("PASSWORD","Rajdip@2001");
+	define("DATABASE","indibook");
+	define("BASEURL", "http://indibook.com");
```
This is not a part of the database configuration so we could separate the line using a blank line.

Eshakundu01 commented on mvc/application/views/register.php:
Proper spacing must be used in the whole file

Rajdip-innoraft755 committed:
FT2023-61-mvc:remove the vendor , bootstrap , jquery files and add .giignore, start working with the landing page .

FT2023-61-mvc:remove the vendor , bootstrap , jquery files and add .giignore, start working with the landing page .

Rajdip-innoraft755 committed:
Merge branch 'main' into FT2023-61-mvc

Rajdip-innoraft755 committed:
FT2023-61-mvc:add more design in landing page , use camel casing in classname, others changes as per mentor's instruction and PR review.

Rajdip-innoraft755 committed:
FT2023-61-mvc:make the landing page dynamic,add feature to post anything and create a sliding menu to user settings.

Eshakundu01 commented on .gitignore:
```diff
@@ -0,0 +1,8 @@
+mvc/vendor/
+*.jpg
+*.png
+*.ico
```
If the images are in a single folder, add the folder path in .gitignore files instead of mentioning individual extensions.

SachinT123 replied:
Although this is fine for the task, please keep in mind that you might need images for some styling purposes but this will ignore all files with the given extensions across all folders. So we should add these files in gitignore.

Rajdip-innoraft755 replied:
Okay , I use  `public/assets/img/` folder for storing the user uploaded image, then should I write in `gitignore` like this `public/assets/img/*jpg` to ignore those images , and store the images for styling in another folder ?

Eshakundu01 commented on mvc/application/controllers/Home.php:
Please correct the spacing in the `home.php`, `landing.php`, and `availableuser.php` files.

Rajdip-innoraft755 committed:
FT2023-61-mvc:add the update profile edit page.

Eshakundu01 commented on mvc/application/models/MakePost.php:
```diff
@@ -0,0 +1,15 @@
+    $this->query($sql);
+    return true;
+  }
+}
```
If leaving a blank line at the beginning of the code make sure to maintain the same at the end of the code.

Rajdip-innoraft755 committed:
FT2023-61-mvc:add the frontend validation in profile page.

Rajdip-innoraft755 committed:
FT2023-61-mvc:fix some bug in register page, add forgotpassword feature using otp, add profile picture image change feature , start adding the feature to post image .

Rajdip-innoraft755 committed:
FT2023-61-mvc: add the feature to add image in post and add config.php in .gitignore.

Rajdip-innoraft755 committed:
FT2023-61-mvc:create dark mode , others users profile view option and a easily accessable navbar in login,register,forgot password page.

Rajdip-innoraft755 committed:
FT2023-61-mvc:add robots.txt , implement the feature to accept/decline cookie policy at the time of login , add the feature to login with google and start implementing the social share option.

Rajdip-innoraft755 committed:
FT2023-61-mvc:there is some problem in git this commit is to solve those.

Rajdip-innoraft755 committed:
FT2023-61-mvc:store cookie in DB at the time of registration and fix a issue in forgot password page.

FT2023-61-mvc:testing.

Revert "FT2023-61-mvc:testing."

This reverts commit ecb66954aace3345f17813b42ae2823b07619f37.

FT2023-61-mvc:fix some gitignore issues.

FT2023-61-mvc:fix issues of gitignore.

Revert "FT2023-61-mvc:fix issues of gitignore."

This reverts commit 7358cca926cf982ec5fb35fb20ae81a37343f041.

Revert "Revert "FT2023-61-mvc:fix issues of gitignore.""

This reverts commit 1e1581aff25a1dcf7ca02194f51aca96c2ba0455.

FT2023-61-mvc:update the composer.json to install vendor .

testing

FT2023-61-mvc: solving git ignore issues.

FT2023-61-mvc: solving git ignore issues.

FT2023-61-mvc:resolving git issues.

solve git isuues.
